### ISSUE TITLE
->text and other similar methods always returns encoded string

### ISSUE THREAD

**Issue by zdm** [Azq2/perl-html5-dom#4]
```
say HTML5::DOM->new->parse('<b>тест</b>')->at('b')->text;
```
As I correctly understand parser detects encoding and store tree in utf-8 internally.
Is it possible to return text and html strings with utf8 flag set, as other html tree builders do (for example HTML::TreeBuilder:: family)?

**Comment by Azq2** [comment]
You means something this?

```perl
#!/usr/bin/perl
use warnings;
use strict;
use HTML5::DOM;
use utf8;
use Encode;

my $parser = HTML5::DOM->new;

my $s = HTML5::DOM->new->parse('<b>пыщь пыщь</b>')->at('b')->text;
Encode::_utf8_on($s);

print $s." ололо\n";
```

Yes, all strings in utf-8. But default behavior - work with bytes. 
I will try implement optional utf8 mode.

Something like this:
utf8 => "auto": if passed source with SvUTF8 == 1, then switch to utf8 mode [default]
utf8 => 1: switch to utf8 mode
uft8 => 0: work with bytes

**Comment by zdm** [comment]
Yes, exactly.
Usually ```->text``` extracted for comparison or concatenation and it is more handy, if it will be encoded to utf8, because this is text string, not a bytes array.
You can use ```sv_utf8_decode(sv)``` - to decode or ```SvUTF8_on(sv)``` to just set utf8 flag (will be faster) in your XS code.

**Comment by Azq2** [comment]
Implemented in 1.14 version. 

Doc:
https://github.com/Azq2/perl-html5-dom#work-with-utf8
https://github.com/Azq2/perl-html5-dom#parser-options
https://github.com/Azq2/perl-html5-dom#utf8

**Comment by zdm** [comment]
Seems, that it works.
One more issue:
according to the spec.  I can set encoding manually in  parser->new:
```
encoding
Encoding of input HTML, if auto - library can tree to automaticaly determine encoding. (default "auto")

Allowed both encoding name or id.
```
but if I specify encoding as string
```
my $parser = HTML5::DOM->new({ encoding => 'WINDOWS-1251' });
```
it produces error:
```
 HTML5::DOM::_parseAsync(): invalid encoding value at /usr/perlbrew/perls/perl-5.28.1/lib/site_perl/5.28.1/x86_64-linux-ld/HTML5/DOM.pm line 321.
```

**Comment by Azq2** [comment]
Fixed in 1.15. Regression test added.

**Comment by Azq2** [comment]
I think problem resolved. Please, open if not.
